### PR TITLE
Added power method for regions

### PR DESCRIPTION
**Pull request by lucbv**
@trilinos/muelu 

## Description
Implementation of a power method to compute estimate of largest eigen value using matrix and vector in region format

## Motivation and Context
This will enable us to add a Chebyshev smoother to region multigrid and potentially help for operator smoothing too.

## Related Issues

* Closes 
* Blocks 
* Is blocked by 
* Follows 
* Precedes 
* Related to 
* Part of 
* Composed of 

## Checklist

- [ ] My commit messages mention the appropriate GitHub issue numbers.
- [ ] My code follows the code style of the affected package(s).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [code contribution guidelines](../blob/master/CONTRIBUTING.md) for this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] No new compiler warnings were introduced.
- [ ] These changes break backwards compatibility.

### CODE REVIEW

**Comment by mhoemmen** [comment]
Alas, yet another reimplementation of the power method... I suppose the algorithm is simple enough that it doesn't pay to repeat, but we might be missing out on future optimizations if we do (I'd like to fuse some kernels in [Ifpack2's implementation](https://github.com/trilinos/Trilinos/blob/c26bf08a4e0994263f31a55bff23d61bd6693dae/packages/ifpack2/src/Ifpack2_Details_Chebyshev_def.hpp#L1343).

**Comment by lucbv** [comment]
@mhoemmen this implementation is using a data structure that cannot be handled by Tpetra or Ifpack2 so it really needs to be custom made...
We are planning on eventually making a push to have our data structures compatible with Tpetra to benefit from all future improvement.
You might notice though that all this work is marked experimental which also probably gives us a little leeway with the rest of production codes.

**Comment by mhoemmen** [comment]
@lucbv I'm lamenting but not asking for changes ;-)

> this implementation is using a data structure that cannot be handled by Tpetra or Ifpack2 so it really needs to be custom made...

Power method just needs an abstract operator and abstract vectors.  Concrete data structures may have opportunities for kernel fusion, but the abstract algorithm should have an abstract implementation.  Just sayin' ;-) .

File: packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp [hunk 1]
```diff
@@ -252,4 +252,98 @@ void smootherApply(RCP<Teuchos::ParameterList> params,
...
+                        const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > mapComp,
+                        const int maxRegPerProc,
+                        const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > rowMapPerGrp,
+                        const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > revisedRowMapPerGrp,
+                        const std::vector<RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> > > rowImportPerGrp)
+{
+#include "Xpetra_UseShortNames.hpp"
+  RCP<Vector> compVec = VectorFactory::Build(mapComp, true);
+  regionalToComposite(regVec, compVec, maxRegPerProc, rowMapPerGrp,
+                      rowImportPerGrp, Xpetra::ADD);
```
[the remark this answers is not in the document]
Comment (pohm01) [reply packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp]: I don't actually know which of the cases we are in, I did check to make sure the result was independent of the number of regions, so it is probably the split value case. Also, I believe `regionalToComposite` is still hard coded to use `Xpetra::ADD` as the CombineMode.

[the remark this answers is not in the document]
Comment (mayrmt) [reply packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp]: > I believe regionalToComposite is still hard coded to use Xpetra::ADD as the CombineMode.

I think that's true. I have added routines to perform scale interface values with 1/numAdjacentRegions. This takes care of this, so we should be fine as far as I know.

**Comment by mayrmt** [comment]
@pohm01 Should we proceed with merging this or will you continue working on this branch? 
@lucbv What's your view on that?

**Comment by lucbv** [comment]
This PR was superseded by PR #5852

**Pull request closed without merging**